src/LinAlg/hiopMatrixMDS.hpp

```diff
@@ -272,16 +281,11 @@ class hiopMatrixSymBlockDiagMDS : public hiopMatrix
   virtual void timesVec(double beta,  hiopVector& y,
 			double alpha, const hiopVector& x) const
   {
-    hiopVectorPar* yp = dynamic_cast<hiopVectorPar*>(&y);
-    const hiopVectorPar* xp = dynamic_cast<const hiopVectorPar*>(&x);
-    assert(yp);
```

Comment:
any reason(s) these asserts were removed?

Reply:
It seems like the dynamic casts were removed to favor using the interface, in which case they're reference types and I don't see the point of keeping those assertions.

Reply:
oh, I see, I misread the code.